### PR TITLE
Reference to pylops in LinearOperator notes and ARPACK example

### PR DESCRIPTION
**Pull request by mrava87**
- Add reference to pylops project in LinearOperator notes. 
- Added example with LinearOperator to Sparse Eigenvalue Problems with ARPACK tutorial.

### CODE REVIEW

**Comment by mrava87** [comment]
I didn't realise that the code snippets in your tutorials are actually tested on travis. How should I go about here:

- add pylops to dependencies for develop (but not user) if this is possible?
- not use pylops in example in ARPACK tutorial but actually create some operators in the tutorial itself (that may make the code longer and quite verbose).
- avoid travis from running the snippet (if possible?)

**Comment by rgommers** [comment]
> not use pylops in example in ARPACK tutorial but actually create some operators in the tutorial itself (that may make the code longer and quite verbose).

This would be preferred. I was thinking something like:
- example of basic `LinearOperator` use
- example of creating an operator (if not too verbose)
- then a short paragraph explaining that PyLops defines more operators and helps also if you want to create your own. and then a link to PyLops.

**Comment by rgommers** [comment]
> add pylops to dependencies for develop (but not user) if this is possible?

this is in principle possible, but I don't think we want to do this. `refguide_check` should also pass locally, so we'd then be requiring devs to also install PyLops.

**Comment by mrava87** [comment]
Hi Ralf,
I agree. I can modify the example as you suggest, should be easy to have one or two operators that are simple to create with few lines of code. Will 
update the pull request :)


iMR

**Comment by mrava87** [comment]
Not sure I understand why Travis is giving all these indentation errors (and just for one specific build)?

Is there a way to locally build the doc with checks on the '>>>' codes? If I run python setup.py build_sphinx everything works fine and the documentation looks like I expect it to look

**Comment by tylerjereddy** [comment]
We have a special script in `tools/refguide_check.py` that executes the docstrings and checks for the expected output, as you've noticed.

In our continuous integration there is a single matrix entry in travis that includes the command `python runtests.py -g --refguide-check`, and you should be able to use that locally as a first pass at refining your docstrings so they pass.

**Comment by mrava87** [comment]
Thanks a lot, that helps! Will do that :)

**Comment by rgommers** [comment]
Don't worry about the PyPy failure, that was just a CI hiccup (`Too long with no output (exceeded 10m0s)`).

File: doc/source/tutorial/arpack.rst [hunk 1]
```diff
@@ -203,6 +205,102 @@ but the operation can also be specified by the user.  See the docstring of
...
+   >>> class Diagonal(LinearOperator):
+   ...     def __init__(self, diag, dtype='float32'):
+   ...         self.diag = diag
+   ...         self.shape = (len(self.diag), len(self.diag))
+   ...         self.dtype = np.dtype(dtype)
+   ...     def _matvec(self, x):
+   ...         return self.diag*x
+   ...     def _rmatvec(self, x):
+   ...         return self.diag*x
+   >>>
```
[the remark this answers is not in the document]
Comment (mrava87) [reply doc/source/tutorial/arpack.rst]: No seems like they can be swapped with blank lines

File: doc/source/tutorial/arpack.rst [hunk 2]
```diff
@@ -203,6 +205,102 @@ but the operation can also be specified by the user.  See the docstring of
...
+          [-0. -1.  0.],
+          [ 0.  0. -1.]]
+
+In this case we have created a quick and easy ``Diagonal`` operator.
+The external library `PyLops <https://pylops.readthedocs.io>`_ provides
+similar capabilities in the `Diagonal <https://pylops.readthedocs.io/en/
+latest/api/generated/pylops.Diagonal.html#pylops.Diagonal>`_ operator
+as well as several other operators.
+
+Finally, we use consider a linear operator that mimics the application of a
```
[the remark this answers is not in the document]
Comment (mrava87) [reply doc/source/tutorial/arpack.rst]: Fixed

File: doc/source/tutorial/arpack.rst [hunk 3]
```diff
@@ -203,6 +205,102 @@ but the operation can also be specified by the user.  See the docstring of
 :func:`scipy.sparse.linalg.eigsh` and
 :func:`scipy.sparse.linalg.eigs` for details.
```
[the remark this answers is not in the document]
Comment (mrava87) [reply doc/source/tutorial/arpack.rst]: Agreed, added it :)

**Comment by mrava87** [comment]
Thanks a lot to you for help :)

**Pull request merged**